src/coreComponents/constitutive/fluid/BlackOilFluid.hpp

```diff
@@ -19,57 +19,1281 @@
 #ifndef GEOSX_CONSTITUTIVE_FLUID_BLACKOILFLUID_HPP_
 #define GEOSX_CONSTITUTIVE_FLUID_BLACKOILFLUID_HPP_
 
-#include "codingUtilities/EnumStrings.hpp"
-#include "constitutive/fluid/MultiFluidPVTPackageWrapper.hpp"
+#include "constitutive/fluid/BlackOilFluidBase.hpp"
+#include "constitutive/fluid/PVTOData.hpp"
+#include "functions/TableFunction.hpp"
+#include "codingUtilities/Utilities.hpp"
 
 namespace geosx
 {
 
 namespace constitutive
 {
 
-class BlackOilFluid : public MultiFluidPVTPackageWrapper
+static constexpr real64 minForPhasePresence = 1e-10;
+
+static constexpr localIndex NC_BO = 3;
+static constexpr localIndex NP_BO = 3;
+static constexpr localIndex HNC_BO = NC_BO - 1;
```

Comment:
I would prefer for these to be scoped in `BlackOilFluid` if they are primarily used in inside that class.